### ISSUE TITLE
`styled()` breaks flow type checking for props?

### ISSUE THREAD

**Issue by sohkai** [styled-components/styled-components#1349]
## Version
```
styled-components: 2.2.3
flow-typed: styled-components_v2.x.x.js
```

## Reproduction

Running flow on something like this results in `styled()` breaking prop type checks:

```jsx
/* @flow */

import type { ComponentType } from 'react'
import React from 'react'
import styled from 'styled-components'

type Props = {
  color: string,
}

const Title: ComponentType<Props> = styled.h1`
  color: ${({ color }) => color};
`

const RenderRedTitle = () => <Title color="red">Title</Title> // Expected -- No flow error
const RenderBadTitle = () => <Title color={3}>Title</Title> // Expected -- Flow error

/**************************
 *     Using `styled()    *
 **************************/
const StyledTitle = styled(Title)`
  font-size: 10px;
`

const RenderRedStyledTitle = () => <StyledTitle color="red">Title</StyledTitle> // Expected -- No flow error
const RenderBadStyledTitle = () => <StyledTitle color={3}>Title</StyledTitle> // Unexpected -- No flow error
```

## Expected Behavior

`const RenderBadStyledTitle = () => <StyledTitle color={3}>Title</StyledTitle>` causes a Flow error on `color` not being a string.

## Actual Behavior

`const RenderBadStyledTitle = () => <StyledTitle color={3}>Title</StyledTitle>` does not cause a Flow error.

**Comment by philpl** [comment]
cc @styled-components/typers

**Comment by sohkai** [comment]
@philpl Ahh, just noticed that there's an actual template in the issue template 😅. I'll reformat :).

**Comment by philpl** [comment]
Moved to #1350

**Comment by sohkai** [comment]
@philpl Hey sorry, #1350 is a different problem than this one, although they look similar (I was running into a few unexpected flow errors all at once). Do you mind reopening this?

**Comment by ryall** [comment]
Same issue in TypeScript. I cannot use `color` as any type other than string - even if the type is defined in my custom props.

**Comment by alejandrodnm** [comment]
@sohkai have you tried doing something like this?

```
const StyledTitle: typeof Title = styled(Title)`
  font-size: 10px;
`
```

**Comment by sohkai** [comment]
@alejandrodnm We switched to not using Flow, but I have not tried that. It does seem like a good workaround for now though.

**Comment by javi10823** [comment]
> @sohkai have you tried doing something like this?
> 
> ```
> const StyledTitle: typeof Title = styled(Title)`
>   font-size: 10px;
> `
> ```

save my life!
